src/anyio/_backends/_asyncio.py

```diff
@@ -320,11 +320,20 @@ def shield(self) -> bool:
         return self._shield
 
 
-async def checkpoint():
+async def checkpoint() -> None:
+    await checkpoint_if_cancelled()
+    await asyncio.sleep(0)
+
+
+async def checkpoint_if_cancelled() -> None:
```

Comment:
trio this is documented as:
```
    This is either a no-op, or else it allow other tasks to be scheduled and
    then raises :exc:`trio.Cancelled`.
```

so it needs an `await asyncio.sleep(0)` somewhere

Reply:
Now it has (I needed to merge the changes from master to make it work).